### ISSUE TITLE
Required parameter $owner follows optional parameter $order at nextcloud/apps/deck/lib/Service/CardService.php#271

### ISSUE THREAD

**Issue by nooblag** [nextcloud/deck#3356]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

**Describe the bug**
`Required parameter $owner follows optional parameter $order at nextcloud/apps/deck/lib/Service/CardService.php#271` error in Nextcloud log.

**To Reproduce**
1. View log

**Expected behavior**
No errors?

**Client details:**
 - OS: Windows 10
 - Browser: Firefox
 - Version: 92.0.1
 - Device: Desktop

<details>
<summary>Server details</summary>

**Operating system**:
Ubuntu 18.04.6 Server LTS

**Web server:**
nginx 1.18.0

**Database:**
PostgreSQL psql 10.18

**PHP version:**
PHP 8.0.11

**Nextcloud version:** (see Nextcloud admin page)
Nextcloud 22.2.0 

**Where did you install Nextcloud from:**
nextcloud.com

**Signing status:**

```
No errors have been found.
```

**List of activated apps:**

```
Enabled:
  - accessibility: 1.8.0
  - activity: 2.15.0
  - admin_audit: 1.12.0
  - announcementcenter: 6.0.0
  - apporder: 0.13.0
  - bruteforcesettings: 2.2.0
  - calendar: 2.3.4
  - camerarawpreviews: 0.7.12
  - circles: 22.1.1
  - cloud_federation_api: 1.5.0
  - cms_pico: 1.0.16
  - comments: 1.12.0
  - dav: 1.19.0
  - deck: 1.5.3
  - drawio: 1.0.1
  - event_update_notification: 1.3.0
  - external: 3.9.0
  - extract: 1.3.2
  - federatedfilesharing: 1.12.0
  - files: 1.17.0
  - files_downloadactivity: 1.11.1
  - files_linkeditor: 1.1.6
  - files_pdfviewer: 2.3.0
  - files_readmemd: 1.2.0
  - files_rightclick: 1.1.0
  - files_sharing: 1.14.0
  - files_texteditor: 2.14.0
  - files_trashbin: 1.12.0
  - files_versions: 1.15.0
  - files_videoplayer: 1.11.0
  - firstrunwizard: 2.11.0
  - forms: 2.3.0
  - groupfolders: 10.0.0
  - impersonate: 1.9.0
  - integration_whiteboard: 0.0.14
  - logreader: 2.7.0
  - lookup_server_connector: 1.10.0
  - metadata: 0.14.0
  - notes: 4.1.1
  - notifications: 2.10.1
  - oauth2: 1.10.0
  - onlyoffice: 7.1.2
  - password_policy: 1.12.0
  - passwords: 2021.10.20
  - photos: 1.4.0
  - privacy: 1.6.0
  - provisioning_api: 1.12.0
  - quota_warning: 1.11.0
  - ransomware_protection: 1.11.0
  - recommendations: 1.1.0
  - serverinfo: 1.12.0
  - settings: 1.4.0
  - sharebymail: 1.12.0
  - systemtags: 1.12.0
  - tasks: 0.14.2
  - text: 3.3.0
  - theming: 1.13.0
  - twofactor_backupcodes: 1.11.0
  - updatenotification: 1.12.0
  - viewer: 1.6.0
  - workflowengine: 2.4.0
Disabled:
  - contactsinteraction
  - dashboard
  - encryption
  - federation
  - files_external
  - nextcloud_announcements
  - previewgenerator
  - support
  - survey_client
  - user_ldap
  - user_status
  - weather_status
```

**Nextcloud configuration:**

```
{
    "system": {
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***",
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "tempdirectory": "***REMOVED SENSITIVE VALUE***\/__temp",
        "skeletondirectory": "***REMOVED SENSITIVE VALUE***\/__skeleton",
        "templatedirectory": "***REMOVED SENSITIVE VALUE***\/__templates",
        "dbtype": "pgsql",
        "version": "22.2.0.2",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "htaccess.RewriteBase": "\/",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "log_type": "file",
        "logfile": "\/var\/log\/nextcloud\/nextcloud.log",
        "loglevel": 2,
        "logtimezone": "Australia\/Melbourne",
        "mail_smtpmode": "smtp",
        "remember_login_cookie_lifetime": 0,
        "log_rotate_size": "10485760",
        "trashbin_retention_obligation": "auto, 14",
        "versions_retention_obligation": "auto, 365",
        "simpleSignUpLink.shown": false,
        "login_form_autocomplete": true,
        "filelocking.enabled": true,
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": "6379",
            "dbindex": "0",
            "timeout": "1.5"
        },
        "maintenance": false,
        "default_phone_region": "AU",
        "default_locale": "en_AU",
        "force_locale": "en_AU",
        "default_language": "en",
        "force_language": "en",
        "defaultapp": "activity,files",
        "activity_expire_days": 180,
        "session_lifetime": 14400,
        "session_keepalive": false,
        "auth.webauthn.enabled": false,
        "overwriteprotocol": "https",
        "app_install_overwrite": [
            "activitylog",
            "files_readmemd",
            "files_texteditor"
        ],
        "updater.release.channel": "stable",
        "theme": "",
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "forwarded_for_headers": [
            "HTTP_CF_CONNECTING_IP"
        ],
        "mail_smtpsecure": "ssl",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
```

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
No

</details>

<details>
<summary>Logs</summary>

#### Nextcloud log (data/nextcloud.log)
```
{"reqId":"obKYzqmT9xEOkWswJBQv","level":3,"time":"2021-10-04T02:41:37+11:00","remoteAddr":"***REMOVED SENSITIVE VALUE***","user":"***REMOVED SENSITIVE VALUE***","app":"PHP","method":"GET","url":"/ocs/v2.php/search/providers?from=%2Fapps%2Factivity%2F","message":"Required parameter $owner follows optional parameter $order at /var/www/nextcloud/apps/deck/lib/Service/CardService.php#271","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:92.0) Gecko/20100101 Firefox/92.0","version":"21.0.5.1","id":"6159dc48209e4"}
```
</details>

**Comment by nooblag** [comment]
Is anything in https://github.com/nextcloud/server/issues/25806 related to this?

**Comment by juliushaertl** [comment]
I cannot reproduce this and the mentioned line has been fixed in the 1.5.3 release that you seem to be running already with #3092 

https://github.com/nextcloud/deck/blob/v1.5.3/lib/Service/CardService.php#L271

Is that the exact release that is running on the instance where you see that error?

**Comment by nooblag** [comment]
Hi @juliushaertl Thanks for getting back. You're correct, I haven't seen that error since upgrading. I'll close this issue.
